### DISCUSSION TITLE
Update all related cache

### DISCUSSION THREAD

**Discussion by pretoricum2** [TanStack/query#3631] [Q&A] [answered]
Hi,

I have a question about keeping all the data in the cache up to date. Best to show it in an example.

Let's say we have personal feed. This feed is made up of FeedItems.
`FeedList:1[FeedItem:1,FeedItem:2,FeedItem:3,FeedItem:4]`

Now let's say there is another FeedList.
`FeedList:2[FeedItem:5,FeedItem:6,FeedItem:1]`

As you can see these lists have `FeedItem:1` in common.

Now let's say we have a detailPage of sorts that we can get by clicking on the FeedItem. In case we fetch the lists in order `FeedList:1` then `FeedList:2`, we can go to the FeedItem detail from FeedList:1 and potentially have old data.

Ideally fetching `FeedList:2` should also make sure that `FeedItem:1` is updated in the first list as well. Is there some solution for this in RQ?

**Comment by TkDodo** [comment, marked as answer]
`invalidateQueries` will make sure that every feed that matches the passed key is either refetched immediately (if used on the screen), or marked as stale so that it will be refetched when it is used the next time. So if you invalidate all list, and then go back to list1, that one will be refetched. If you go to list2, that will be refetched.

You can also manually update all feeds in the onSuccess callback if one of the feeds fetches with `queryClient.setQueriesData`, but I don't think it's necessary.

If you're looking for something more automatic, you basically want something like a normalized cache, where you also need a schema, which is something that react-query does not require, thus it does not provide a normalized cache. GraphQL specific solutions are good at doing that (apollo, urql).

**Reply by pretoricum2** [reply]
Thank you, I was afraid you were going to mention graphQL.
